Build errors with Short and Open

Using Short or Open generates a build error:

chowdsp_wdf\include\chowdsp_wdf\wdf\wdf_nonlinearities.h(76,19): error C2039: 'b': is not a member of 'chowdsp::w
df::Short<double>' [build\TD_Model.vcxproj]
  (compiling source file '../src/main.cpp')
      third_party\chowdsp_wdf\include\chowdsp_wdf\wdf\wdf_nonlinearities.h(56,11):
      see declaration of 'chowdsp::wdf::Short<double>'
      third_party\chowdsp_wdf\include\chowdsp_wdf\wdf\wdf_nonlinearities.h(76,19):
      the template instantiation context (the oldest one first) is
          src\DiodeClipper.h(80,37):
          see reference to class template instantiation 'chowdsp::wdf::Short<double>' being compiled
          third_party\chowdsp_wdf\include\chowdsp_wdf\wdf\wdf_nonlinearities.h(74,18):
          while compiling class template member function 'T chowdsp::wdf::Short<T>::reflected(void) noexcept'
          with
          [
              T=double
          ]

third_party\chowdsp_wdf\include\chowdsp_wdf\wdf\wdf_nonlinearities.h(76,30): error C2039: 'a': is not a member of 'chowdsp::w 
df::Short<double>' [build\TD_Model.vcxproj]
  (compiling source file '../src/main.cpp')
      third_party\chowdsp_wdf\include\chowdsp_wdf\wdf\wdf_nonlinearities.h(56,11):
      see declaration of 'chowdsp::wdf::Short<double>'

third_party\chowdsp_wdf\include\chowdsp_wdf\wdf\wdf_nonlinearities.h(77,26): error C2039: 'b': is not a member of 'chowdsp::w 
df::Short<double>' [build\TD_Model.vcxproj]
  (compiling source file '../src/main.cpp')
      third_party\chowdsp_wdf\include\chowdsp_wdf\wdf\wdf_nonlinearities.h(56,11):
      see declaration of 'chowdsp::wdf::Short<double>'

--- a/.github/workflows/bench.yml
+++ b/.github/workflows/bench.yml
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false # show all errors for each platform (vs. cancel jobs on error)
       matrix:
-        os: [ubuntu-latest, windows-2019, macos-latest]
+        os: [ubuntu-latest, windows-2022, macos-latest]
 
     steps:
       - name: Get latest CMake

--- a/.github/workflows/bench.yml
+++ b/.github/workflows/bench.yml
@@ -28,7 +28,7 @@ jobs:
 
       - name: Configure
         shell: bash
-        run: cmake -DCMAKE_BUILD_TYPE=Release -DCHOWDSP_WDF_TEST_WITH_XSIMD_VERSION=8.0.5 -Bbuild
+        run: cmake -DCMAKE_BUILD_TYPE=Release -DCHOWDSP_WDF_TEST_WITH_XSIMD_VERSION="13.2.0" -Bbuild
 
       - name: Build
         shell: bash

--- a/.github/workflows/cov.yml
+++ b/.github/workflows/cov.yml
@@ -49,8 +49,8 @@ jobs:
       - name: Collect Coverage Data
         shell: bash
         run: |
-          lcov --directory . --capture --output-file coverage.info
-          lcov --remove coverage.info '/usr/*' "${HOME}"'/.cache/*' '/Applications/Xcode*' '*tests*' --output-file coverage.info
+          lcov --ignore-errors mismatch --directory . --capture --output-file coverage.info
+          lcov --remove coverage.info '/usr/*' "${HOME}"'/.cache/*' '/Applications/Xcode*' '*chowdsp_wdf/build/*' '*chowdsp_wdf/tests/*' --output-file coverage.info
 
       - name: Report Coverage Data
         shell: bash

--- a/.github/workflows/cov.yml
+++ b/.github/workflows/cov.yml
@@ -18,7 +18,7 @@ jobs:
     strategy:
       fail-fast: false # show all errors for each platform (vs. cancel jobs on error)
       matrix:
-        xsimd_version: ["", "8.0.5"]
+        xsimd_version: ["", "13.2.0"]
 
     steps:
       - name: Install Linux Deps

--- a/.github/workflows/cov.yml
+++ b/.github/workflows/cov.yml
@@ -50,7 +50,7 @@ jobs:
         shell: bash
         run: |
           lcov --ignore-errors mismatch --directory . --capture --output-file coverage.info
-          lcov --remove coverage.info '/usr/*' "${HOME}"'/.cache/*' '/Applications/Xcode*' '*chowdsp_wdf/build/*' '*chowdsp_wdf/tests/*' --output-file coverage.info
+          lcov --ignore-errors unused --remove coverage.info '/usr/*' "${HOME}"'/.cache/*' '/Applications/Xcode*' '*chowdsp_wdf/build/*' '*chowdsp_wdf/tests/*' --output-file coverage.info
 
       - name: Report Coverage Data
         shell: bash

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -17,8 +17,8 @@ jobs:
     strategy:
       fail-fast: false # show all errors for each platform (vs. cancel jobs on error)
       matrix:
-        os: [ubuntu-latest, windows-2019, macos-latest]
-        xsimd_version: ["", "9.0.1", "8.0.0", "8.1.0"]
+        os: [ubuntu-latest, windows-2022, macos-latest]
+        xsimd_version: ["", "12.1.1", "13.2.0"]
         include:
           - os: macos-latest
             xsimd_version: "8.0.5"

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -21,7 +21,7 @@ jobs:
         xsimd_version: ["", "12.1.1", "13.2.0"]
         include:
           - os: macos-latest
-            xsimd_version: "8.0.5"
+            xsimd_version: "13.2.0"
             cmake_args: "-GXcode -D\"CMAKE_OSX_ARCHITECTURES=arm64;x86_64\" -DCHOWDSP_WDF_BUILD_BENCHMARKS=OFF"
 
     steps:

--- a/include/chowdsp_wdf/wdf/wdf_nonlinearities.h
+++ b/include/chowdsp_wdf/wdf/wdf_nonlinearities.h
@@ -31,8 +31,8 @@ namespace wdf
     public:
         Open() : WDF<T> ("Open")
         {
-            this->R = (T) 1.0e15;
-            this->G = (T) 1.0 / this->R;
+            this->wdf.R = (T) 1.0e15;
+            this->wdf.G = (T) 1.0 / this->wdf.R;
         }
 
         inline void calcImpedance() override {}
@@ -40,14 +40,14 @@ namespace wdf
         /** Accepts an incident wave into a WDF open. */
         inline void incident (T x) noexcept override
         {
-            this->a = x;
+            this->wdf.a = x;
         }
 
         /** Propogates a reflected wave from a WDF open. */
         inline T reflected() noexcept override
         {
-            this->b = this->a;
-            return this->b;
+            this->wdf.b = this->wdf.a;
+            return this->wdf.b;
         }
     };
 
@@ -58,8 +58,8 @@ namespace wdf
     public:
         Short() : WDF<T> ("Short")
         {
-            this->R = (T) 1.0e-15;
-            this->G = (T) 1.0 / this->R;
+            this->wdf.R = (T) 1.0e-15;
+            this->wdf.G = (T) 1.0 / this->wdf.R;
         }
 
         inline void calcImpedance() override {}
@@ -67,14 +67,14 @@ namespace wdf
         /** Accepts an incident wave into a WDF short. */
         inline void incident (T x) noexcept override
         {
-            this->a = x;
+            this->wdf.a = x;
         }
 
         /** Propogates a reflected wave from a WDF short. */
         inline T reflected() noexcept override
         {
-            this->b = -this->a;
-            return this->b;
+            this->wdf.b = -this->wdf.a;
+            return this->wdf.b;
         }
     };
 

--- a/tests/BasicCircuitTest.cpp
+++ b/tests/BasicCircuitTest.cpp
@@ -281,4 +281,28 @@ TEST_CASE ("Basic Circuits Test")
         doImpedanceChecks (
             ResistiveCurrentSource<float> { 1000.0f }, 1000.0f, 2000.0f, [=] (auto& r, float value) { r.setResistanceValue (value); }, [=] (float value) { return value; });
     }
+
+    SECTION ("Open Test")
+    {
+        ResistiveVoltageSource<float> Vin { 1.0e3f };
+        Open<float> open {};
+
+        Vin.setVoltage (10.0f);
+        open.incident (Vin.reflected());
+        Vin.incident (open.reflected());
+
+        REQUIRE (open.voltage() == 10.0f);
+    }
+
+    SECTION ("Short Test")
+    {
+        ResistiveVoltageSource<float> Vin { 1.0e3f };
+        Short<float> sh {};
+
+        Vin.setVoltage (10.0f);
+        sh.incident (Vin.reflected());
+        Vin.incident (sh.reflected());
+
+        REQUIRE (sh.voltage() == 0.0f);
+    }
 }

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,6 +1,7 @@
 add_executable(chowdsp_wdf_tests)
 target_include_directories(chowdsp_wdf_tests PRIVATE .)
 target_link_libraries(chowdsp_wdf_tests PRIVATE ${PROJECT_NAME} chowdsp_wdf)
+target_compile_definitions(chowdsp_wdf_tests PRIVATE _USE_MATH_DEFINES=1)
 target_sources(chowdsp_wdf_tests
     PRIVATE
         BasicCircuitTest.cpp

--- a/tests/CombinedComponentTest.cpp
+++ b/tests/CombinedComponentTest.cpp
@@ -6,81 +6,81 @@ using namespace chowdsp::wdft;
 
 TEST_CASE ("Combined Component Test")
 {
-    //    SECTION ("Resistor/Capacitor Series")
-    //    {
-    //        static constexpr auto r_val = 2000.0f;
-    //        static constexpr auto c_val = 2.0e-6f;
-    //
-    //        ResistorT<float> r1 { r_val };
-    //        CapacitorT<float> c1 { c_val };
-    //        WDFSeriesT<float, decltype (r1), decltype (c1)> s1 { r1, c1 };
-    //
-    //        ResistorCapacitorSeriesT<float> rc1 { r_val, c_val };
-    //
-    //        float inputs[] = { 0.0f, 1.0f, -1.0f, 2.0f, -3.0f };
-    //        for (auto& a : inputs)
-    //        {
-    //            s1.incident (a);
-    //            const auto ref = s1.reflected();
-    //
-    //            rc1.incident (a);
-    //            const auto actual = rc1.reflected();
-    //
-    //            REQUIRE (ref == Approx { actual }.margin (1.0e-4f));
-    //        }
-    //    }
-    //
-    //    SECTION ("Resistor/Capacitor Parallel")
-    //    {
-    //        static constexpr auto r_val = 2000.0f;
-    //        static constexpr auto c_val = 2.0e-6f;
-    //
-    //        ResistorT<float> r1 { r_val };
-    //        CapacitorT<float> c1 { c_val };
-    //        WDFParallelT<float, decltype (r1), decltype (c1)> p1 { r1, c1 };
-    //
-    //        ResistorCapacitorParallelT<float> rc1 { r_val, c_val };
-    //
-    //        float inputs[] = { 0.0f, 1.0f, -1.0f, 2.0f, -3.0f };
-    //        for (auto& a : inputs)
-    //        {
-    //            p1.incident (a);
-    //            const auto ref = p1.reflected();
-    //
-    //            rc1.incident (a);
-    //            const auto actual = rc1.reflected();
-    //
-    //            REQUIRE (ref == Approx { actual }.margin (1.0e-4f));
-    //        }
-    //    }
-    //
-    //    SECTION ("Resistor/Capacitor/Voltage Source Series")
-    //    {
-    //        static constexpr auto r_val = 2000.0f;
-    //        static constexpr auto c_val = 2.0e-6f;
-    //        static constexpr auto source_v = 1.5f;
-    //
-    //        ResistiveVoltageSourceT<float> rv1 { r_val };
-    //        rv1.setVoltage (source_v);
-    //        CapacitorT<float> c1 { c_val };
-    //        WDFSeriesT<float, decltype (rv1), decltype (c1)> s1 { rv1, c1 };
-    //
-    //        ResistiveCapacitiveVoltageSourceT<float> rc1 { r_val, c_val };
-    //        rc1.setVoltage (source_v);
-    //        rc1.reset();
-    //
-    //        float inputs[] = { 0.0f, 1.0f, -1.0f, 2.0f, -3.0f };
-    //        for (auto& a : inputs)
-    //        {
-    //            s1.incident (a);
-    //            const auto ref = s1.reflected();
-    //
-    //            rc1.incident (a);
-    //            const auto actual = rc1.reflected();
-    //
-    //            REQUIRE (ref == Approx { actual }.margin (1.0e-4f));
-    //        }
-    //    }
+    SECTION ("Resistor/Capacitor Series")
+    {
+        static constexpr auto r_val = 2000.0f;
+        static constexpr auto c_val = 2.0e-6f;
+
+        ResistorT<float> r1 { r_val };
+        CapacitorT<float> c1 { c_val };
+        WDFSeriesT<float, decltype (r1), decltype (c1)> s1 { r1, c1 };
+
+        ResistorCapacitorSeriesT<float> rc1 { r_val, c_val };
+
+        float inputs[] = { 0.0f, 1.0f, -1.0f, 2.0f, -3.0f };
+        for (auto& a : inputs)
+        {
+            s1.incident (a);
+            const auto ref = s1.reflected();
+
+            rc1.incident (a);
+            const auto actual = rc1.reflected();
+
+            REQUIRE (ref == Approx { actual }.margin (1.0e-4f));
+        }
+    }
+
+    SECTION ("Resistor/Capacitor Parallel")
+    {
+        static constexpr auto r_val = 2000.0f;
+        static constexpr auto c_val = 2.0e-6f;
+
+        ResistorT<float> r1 { r_val };
+        CapacitorT<float> c1 { c_val };
+        WDFParallelT<float, decltype (r1), decltype (c1)> p1 { r1, c1 };
+
+        ResistorCapacitorParallelT<float> rc1 { r_val, c_val };
+
+        float inputs[] = { 0.0f, 1.0f, -1.0f, 2.0f, -3.0f };
+        for (auto& a : inputs)
+        {
+            p1.incident (a);
+            const auto ref = p1.reflected();
+
+            rc1.incident (a);
+            const auto actual = rc1.reflected();
+
+            REQUIRE (ref == Approx { actual }.margin (1.0e-4f));
+        }
+    }
+
+    SECTION ("Resistor/Capacitor/Voltage Source Series")
+    {
+        static constexpr auto r_val = 2000.0f;
+        static constexpr auto c_val = 2.0e-6f;
+        static constexpr auto source_v = 1.5f;
+
+        ResistiveVoltageSourceT<float> rv1 { r_val };
+        rv1.setVoltage (source_v);
+        CapacitorT<float> c1 { c_val };
+        WDFSeriesT<float, decltype (rv1), decltype (c1)> s1 { rv1, c1 };
+
+        ResistiveCapacitiveVoltageSourceT<float> rc1 { r_val, c_val };
+        rc1.setVoltage (source_v);
+        rc1.reset();
+
+        float inputs[] = { 0.0f, 1.0f, -1.0f, 2.0f, -3.0f };
+        for (auto& a : inputs)
+        {
+            s1.incident (a);
+            const auto ref = s1.reflected();
+
+            rc1.incident (a);
+            const auto actual = rc1.reflected();
+
+            REQUIRE (ref == Approx { actual }.margin (1.0e-4f));
+        }
+    }
 
     SECTION ("Capacitive Voltage Source")
     {

